sqlglot/transforms.py

```diff
@@ -628,11 +628,30 @@ def eliminate_full_outer_join(expression: exp.Expression) -> exp.Expression:
             expression_copy = expression.copy()
             expression.set("limit", None)
             index, full_outer_join = full_outer_joins[0]
+
+            tables = (
+                full_outer_join.alias_or_name,
+                expression.args["from"].alias_or_name,
+            )
+            join_conditions = full_outer_join.args.get("on") or exp.and_(
+                *[
+                    exp.column(col, tables[0]).eq(exp.column(col, tables[1]))
+                    for col in full_outer_join.args.get("using")
```

Comment:
Do we have a test case for `USING`?

Reply:
If there's an `ON` condition do we still need to add this?

Reply:
I'll add it

Reply:
I added a test case for USING since it has its own logic in `eliminate_full_outer_join`, if it's not needed I'll revert it

Reply:
I meant that if you have an `ON` clause, then you skip the `exp.and_` term. What if that `ON` clause doesn't have the equality predicate? Does this transformation still work then, or do we still need to append an `AND` term to that `ON`?

Reply:
So the `exp.and_` is needed only when dealing with `USING` because we need to transform the using condition into an implicit join to feed into the anti-join condition: `NOT EXISTS (SELECT 1 FROM <left_side_table> WHERE <join_condition_here>)`.

If we have an `ON` clause instead, it should be already compatible and ready to be used in later steps, even if it is a pure non equi join, I have attached a real example with this behaviour.
![image](https://github.com/user-attachments/assets/c3cbd71f-3b8c-488d-b951-c84bd69b08ac)

Let me know if you think I'm missing something

Reply:
So the `exp.and_` is needed only when dealing with `USING` because we need to transform the using condition into an implicit join to feed into the anti-join condition: `NOT EXISTS (SELECT 1 FROM <left_side_table> WHERE <join_condition_here>)`.

If we have an `ON` clause instead, it should be already compatible and ready to be used in later steps, even if it is a pure non equi join, I have attached a real example with this behaviour.
![image](https://github.com/user-attachments/assets/fd6e6c81-be71-4f73-bd5d-fa7a96c7f8ff)


Let me know if you think I'm missing something

Reply:
I see, this makes sense. Thanks for clarifying!